### PR TITLE
comm/tcp listener: do not pass comm with failed handshake to comm_handler

### PR DESCRIPTION
**Pull request by jochen-ott-by**
A return was missing here, which would pass on `comm` to the handler, even if the handshake had errors.

### CODE REVIEW

**Review by fjetter — APPROVED** [review]
Thanks @jochen-ott-by for spotting this!

Will wait a bit to give other maintainers the possibility to chime in before I merge but I think this is a safe thing

**Comment by fjetter** [comment]
Both py3.7/3.8 fail in `test_dynamic_workloads_sync` which doesn't seem to be a known flaky test

**Comment by jochen-ott-by** [comment]
> Both py3.7/3.8 fail in `test_dynamic_workloads_sync` which doesn't seem to be a known flaky test

They fail only on windows. Not sure how to debug this tbh; I don't have access to a Windows machine here and the tests run fine for me on my linux machine for python3.6 and python3.8. I'd need some advice here from someone more familiar with the test setup and the differences between Windows and Linux that may come in here.

**Comment by mrocklin** [comment]
Restarting Windows CI to see if they fail differently next time

It looks like `test_broken_worker_during_computation` failed on Travis/Linux

**Comment by mrocklin** [comment]
Thank you for finding this @jochen-ott-by

**Comment by fjetter** [comment]
`test_broken_worker_during_computation` is a known issue https://github.com/dask/distributed/issues/4173

**Comment by jochen-ott-by** [comment]
> Restarting Windows CI to see if they fail differently next time

Indeed, they did: the first time, python3.7/python3.8 had errors while python3.6 was fine. The second run was the the exact opposite: python3.6 had errors (not test failure, only 1 error: "ERROR at teardown of test_get_task_stream_save"), python3.7/python3.8 were fine. So we've seen exactly one green build for each of python3.6/3.7/3.8 on windows.

**Comment by fjetter** [comment]
I tried restarting the GH actions one more time since I haven't seen the other breaking test anywhere, yet. However, the github security changes now break the builds. The fix is already on master. @jochen-ott-by can you please rebase once more?

File: distributed/comm/tests/test_comms.py
Comment (jrbourbeau) [inline distributed/comm/tests/test_comms.py:219]: I'm curious what prompted the addition of this `asyncio.sleep`

**Review by jrbourbeau — COMMENTED** [review]
Thanks for your work on this @jochen-ott-by! Just a couple of small comments

File: distributed/comm/tests/test_comms.py
Comment (jochen-ott-by) [reply distributed/comm/tests/test_comms.py:219]: This ensures we actually hit the timeout of the handshake path on the server side. Without this, the test would be useless. I'll add a comment.

File: distributed/comm/tests/test_comms.py
Comment (jochen-ott-by) [reply distributed/comm/tests/test_comms.py:224]: I added it in the second commit, if the `wait_closed` method exists (it does not for python3.6).

File: distributed/comm/tests/test_comms.py
Comment (lr4d) [inline distributed/comm/tests/test_comms.py:219]:
```suggestion
        # wait a bit to let the listener side hit the timeout on the handshake:
```

**Review by lr4d — APPROVED** [review]

**Review by jrbourbeau — APPROVED** [review]
Thanks for the fix @jochen-ott-by! And @fjetter @lr4d for reviewing

**Pull request merged**
